CDTDatastore/CDTReplicator/CDTPullReplication.m

```diff
@@ -33,7 +34,16 @@ + (instancetype)replicationWithSource:(NSURL *)source target:(CDTDatastore *)tar
 - (instancetype)initWithSource:(NSURL *)source target:(CDTDatastore *)target
 {
     if (self = [super init]) {
-        _source = source;
+        
+        NSURLComponents * sourceComponents = [NSURLComponents componentsWithURL:source resolvingAgainstBaseURL:NO];
```

Comment:
Using the init would lead to more duplication because the urls are stored as different properties, and they would still need to be stripped of creds. 

Using two sets of methods would give us similar duplication that we have no, since we'd need a method to create the interceptor, which would break the URL into components, and then extract the username and password if provided. The URL method would need to break it into components and then remove the user/password and return the url. 

If this was pre-1.0 we attempt something using the generics, but I think doing that now might result in 1.1 breaking semver, but I'm not sure on that.